core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/codeGen/MarkersExtractor.kt

```diff
@@ -17,14 +17,33 @@ import kotlin.reflect.full.withNullability
 import kotlin.reflect.jvm.jvmErasure
 import kotlin.reflect.typeOf
 
-internal fun KType.shouldBeConvertedToFrameColumn(): Boolean = when (jvmErasure) {
-    DataFrame::class -> true
-    List::class -> arguments[0].type?.jvmErasure?.hasAnnotation<DataSchema>() == true
-    else -> false
+internal fun KType.getFieldKind(): FieldKind = when {
+    jvmErasure == DataFrame::class -> Frame
+    jvmErasure == List::class && (arguments[0].type?.jvmErasure?.hasAnnotation<DataSchema>() == true) -> ListToFrame
+    jvmErasure == DataRow::class -> Group
+    jvmErasure.hasAnnotation<DataSchema>() -> ObjectToGroup
+    else -> Default
 }
 
-internal fun KType.shouldBeConvertedToColumnGroup(): Boolean = jvmErasure.let {
-    it == DataRow::class || it.hasAnnotation<DataSchema>()
+internal sealed interface FieldKind {
+    val shouldBeConvertedToColumnGroup: Boolean get() = false
+    val shouldBeConvertedToFrameColumn: Boolean get() = false
+}
+internal data object Frame : FieldKind {
+    override val shouldBeConvertedToFrameColumn: Boolean = true
+}
+internal data object ListToFrame : FieldKind {
+    override val shouldBeConvertedToFrameColumn: Boolean = true
+}
+
+internal data object Default : FieldKind
```

Comment:
It will be a good to explain the bunch of these descendants of FieldKind

Reply:
1 for each branch of getFieldKind :) Previously this function returned just Boolean, but for rendering purposes we need to return more info.